src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -967,17 +967,12 @@ where
                     }
                 },
                 _ = work_to_do.notified(), if timestamper.initialized() => {
-                    // The frontier at which we can retire work is the minimum of the timestamper
-                    // upper and the source data upper
-                    let mut ready_upper = timestamper.source_upper();
-                    ready_upper.extend(source_upper.frontier().iter().cloned());
-
                     // Drain all messages that can be reclocked from all the batches
                     let total_buffered: usize = untimestamped_batches.iter().map(|b| b.len()).sum();
                     let msgs = untimestamped_batches
                         .iter_mut()
                         .flat_map(|batch| batch.drain_filter_swapping(|(_, ts, _)| {
-                            !ready_upper.less_equal(ts)
+                            !timestamper.source_upper().less_equal(ts)
```

Comment:
ah yes, as soon as we have the remap information we can reclock and retire, right?

Reply:
Yes